File: apps/app-frontend/src/helpers/background.ts
import { convertFileSrc, invoke } from '@tauri-apps/api/core'

/**
 * Copies a user-chosen image into the launcher's backgrounds cache (downscaled, re-encoded) and
 * returns the path of the copy, to be stored in a `BackgroundConfig` of type `image`.
 */
export async function cacheBackgroundImage(path: string): Promise<string> {
	return await invoke('plugin:background|background_cache_image', { path })
}

export function getBackgroundImageUrl(path: string): string {
	return convertFileSrc(path)
}

File: apps/app/src/api/background.rs
use crate::api::Result;
use std::path::Path;
use theseus::prelude::*;

pub fn init<R: tauri::Runtime>() -> tauri::plugin::TauriPlugin<R> {
    tauri::plugin::Builder::new("background")
        .invoke_handler(tauri::generate_handler![background_cache_image])
        .build()
}

// Copy a user-chosen image into the launcher's backgrounds cache and return the path of the copy
// invoke('plugin:background|background_cache_image', { path })
#[tauri::command]
pub async fn background_cache_image(path: &Path) -> Result<String> {
    Ok(background::cache_image(path).await?)
}
